src/Mapper/DatabaseMapper.php

```diff
@@ -197,11 +190,7 @@ protected function mapColumns(array $columns): array
     {
         $result = [];
         foreach ($columns as $column => $value) {
-            if (array_key_exists($column, $this->columns)) {
-                $result[$this->columns[$column]] = $value;
-            } else {
-                $result[$column] = $value;
-            }
+            $result[$this->columns[$column] ?? $column] = $value;
```

Comment:
What if column value is null?

Reply:
In case the key exists with a null value (i.e. `$this->columns[$column] === null`):
1. previous code results in `$result[null] = $value`, i.e. `$result[''] = $value` (?)
2. my change results in: `$result[$column] = $value;`

in both cases what is fed into `mapColumns` has already been filtered in both via `fetchFields` which intersect the keys usingthe keys of `$this->fields`, which in turn is derived from `$this->columns`. So at least in these cases `$column` should be either db-column name or an entity-property name.
I am not sure if mapColumns is used elsewhere and if any key that is not a db-column or a mapped entity property is used to insert/update related entities. If not we maybe should further restrict the mapColumns resulting array, discarding any non-db-column key, like this:
```php
    protected function mapColumns(array $data): array
    {
        $result = [];
        foreach ($data as $column => $value) {
            if (isset($this->columns[$column])) {
                $result[$this->columns[$column]] = $value;
            } elseif (in_array($column, $this->columns, true)) { // (*)
                $result[$column] = $value;
            }
        }

        return $result;
    }
```

I'd also suggest another change when defining the fields:
```php
$this->fields[$column] = is_numeric($name) ? $column : $name;
```
so that a field's key is its related column (or itself if identical). This helps us to avoid using `in_array($key, $array, true)` which is slower than `isset($array[$key])` (*)

kind regards

Reply:
Not sure if restriction needed for now, this method is mostly for legacy notation where only field names are specified. By now all most of schemas generated automatically or as per docs include both field and column. Thank you so much for your PR